chain/solana/src/chain.rs

```diff
@@ -0,0 +1,299 @@
+        &self,
+        _loc: &DeploymentLocator,
+        _capabilities: &Self::NodeCapabilities,
+        _unified_api_version: UnifiedMappingApiVersion,
+        _stopwatch_metrics: StopwatchMetrics,
+    ) -> Result<Arc<Self::TriggersAdapter>, Error> {
+        let adapter = TriggersAdapter {};
+        Ok(Arc::new(adapter))
+    }
+
+    async fn new_firehose_block_stream(
+        &self,
+        deployment: DeploymentLocator,
+        store: Arc<dyn WritableStore>,
+        start_blocks: Vec<BlockNumber>,
+        filter: Arc<Self::TriggerFilter>,
+        metrics: Arc<BlockStreamMetrics>,
+        unified_api_version: UnifiedMappingApiVersion,
+    ) -> Result<Box<dyn BlockStream<Self>>, Error> {
+        let adapter = self
+            .triggers_adapter(
+                &deployment,
+                &NodeCapabilities {},
+                unified_api_version.clone(),
+                metrics.stopwatch.clone(),
+            )
+            .expect(&format!("no adapter for network {}", self.name,));
+
+        let firehose_endpoint = match self.firehose_endpoints.random() {
+            Some(e) => e.clone(),
+            None => return Err(anyhow::format_err!("no firehose endpoint available",)),
+        };
+
+        let logger = self
+            .logger_factory
+            .subgraph_logger(&deployment)
+            .new(o!("component" => "FirehoseBlockStream"));
+
+        let firehose_mapper = Arc::new(FirehoseMapper {});
+        let firehose_cursor = store.block_cursor();
+
+        Ok(Box::new(FirehoseBlockStream::new(
+            firehose_endpoint,
+            firehose_cursor,
+            firehose_mapper,
+            adapter,
+            filter,
+            start_blocks,
+            logger,
+        )))
+    }
+
+    async fn new_polling_block_stream(
+        &self,
+        _deployment: DeploymentLocator,
+        _writable: Arc<dyn WritableStore>,
+        _start_blocks: Vec<BlockNumber>,
+        _subgraph_start_block: Option<BlockPtr>,
+        _filter: Arc<Self::TriggerFilter>,
+        _metrics: Arc<BlockStreamMetrics>,
+        _unified_api_version: UnifiedMappingApiVersion,
+    ) -> Result<Box<dyn BlockStream<Self>>, Error> {
+        panic!("SOLANA does not support polling block stream")
+    }
+
+    fn chain_store(&self) -> Arc<dyn ChainStore> {
+        self.chain_store.clone()
+    }
+
+    async fn block_pointer_from_number(
+        &self,
+        _logger: &Logger,
+        _number: BlockNumber,
+    ) -> Result<BlockPtr, IngestorError> {
+        // FIXME (Solana): Hmmm, what to do with this?
+        Ok(BlockPtr {
+            hash: BlockHash::from(vec![0xff; 32]),
+            number: 0,
+        })
+    }
+
+    fn runtime_adapter(&self) -> Arc<Self::RuntimeAdapter> {
+        Arc::new(RuntimeAdapter {})
+    }
+
+    fn is_firehose_supported(&self) -> bool {
+        true
+    }
+}
+
+pub struct TriggersAdapter {}
+
+#[async_trait]
+impl TriggersAdapterTrait<Chain> for TriggersAdapter {
```

Comment:
See new NEAR implementation to correctly set up those methods: https://github.com/graphprotocol/graph-node/blob/master/chain/near/src/chain.rs#L183